drivers/led_strip/ws2812_spi.c

```diff
@@ -197,15 +200,25 @@ static const struct led_strip_driver_api ws2812_spi_api = {
 /* Get the latch/reset delay from the "reset-delay" DT property. */
 #define WS2812_RESET_DELAY(idx) \
 	(DT_INST_PROP(idx, reset_delay))
-#if IS_ENABLED(CONFIG_WS2812_STRIP_SPI_PREAMBLE_SUPPORT)
-#define WS2812_PREAMBLE_FRAMES(idx) \
+#if IS_ENABLED(CONFIG_WS2812_STRIP_SPI_PREAMBLE_SUPPORT) || IS_ENABLED(CONFIG_WS2812_STRIP_SPI_POSTAMBLE_SUPPORT)
+#define WS2812_PADDING_FRAMES(idx) \
+	((DT_INST_PROP(idx, spi_max_frequency) * WS2812_RESET_DELAY(idx))/8000000)
+#else
+#define WS2812_PADDING_FRAMES(idx) \
+	(0)
+#endif
+#if IS_ENABLED(CONFIG_WS2812_STRIP_SPI_POSTAMBLE_SUPPORT)
+#define WS2812_POSTAMBLE_FRAMES(idx) \
 	((DT_INST_PROP(idx, spi_max_frequency) * WS2812_RESET_DELAY(idx))/8000000)
 #else
-#define WS2812_PREAMBLE_FRAMES(idx) \
+#define WS2812_POSTAMBLE_FRAMES(idx) \
 	(0)
 #endif
+/* size of buffer is number of pixel bits + pre/postamble padding if enabled */
 #define WS2812_SPI_BUFSZ(idx) \
-	(WS2812_PREAMBLE_FRAMES(idx) + (WS2812_NUM_COLORS(idx) * 8 * WS2812_SPI_NUM_PIXELS(idx)))
+	((WS2812_PADDING_FRAMES(idx) \
+    * (IS_ENABLED(CONFIG_WS2812_STRIP_SPI_PREAMBLE_SUPPORT) + IS_ENABLED(CONFIG_WS2812_STRIP_SPI_POSTAMBLE_SUPPORT))) \
+    + (WS2812_NUM_COLORS(idx) * 8 * WS2812_SPI_NUM_PIXELS(idx)))
```

Comment:
So the total extra ram consumption for this feature is:
2 bytes for holding the calculated pad length
(SPI_FREQUENCY * RESET_DELAY / 8,000,000) Bytes for each amble (so max 2x)
In practice the total additional RAM usage for a typical IMXRT setup is 77 bytes. 
IMO this level of memory overhead for a processor typically configured with MB scale memories is more then reasonable. As it stand the IMXRT platform is the only one with this issue.

As a general statement I would say this feature uses too much memory, but unfortunately this is the only practical way I can see the WS2812B SPI driver operating on NXPs IMXRTxxxx LPSPI system without wider butchering. 
This will remain completely compiled out when being used on normal behaviour SPI platforms.